src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -457,20 +461,45 @@ dnnl::post_ops program_node::try_optimize_post_ops(dnnl::post_ops& p_ops, const
         }
     };
 
+    auto remove_optimized_prefix = [&](std::vector<fused_primitive_desc_onednn>& post_ops) {
+        // Check and update post-op map if we already optimized something
+        auto iter = post_ops.begin();
+        for (; iter != post_ops.end(); ) {
```

Comment:
nit: What about `while` instead of `for`?

Reply:
It seems to be good.